commerce_coordinator/apps/ecommerce/views.py

```diff
@@ -12,30 +11,40 @@
 
 from .signals import enrollment_code_redemption_requested_signal
 
-logger = logging.getLogger(__name__)
-
 
 class RedeemEnrollmentCodeView(APIView):
     """User requests to redeem enrollment code."""
     permission_classes = [LoginRedirectIfUnauthenticated]
     throttle_classes = [UserRateThrottle]
 
     def get(self, request):
-        """Return paginated response of user's order history."""
+        """
+        Redeem an enrollment code for an authenticated user.
+
+        Args:
+            sku: ecommerce partner_sku to be redeemed
+            code: enrollment code (aka, 100 percent off coupon code) to redeem.
+
+        Returns:
+            Dictionary with results from signal dispatch to redeem an enrollment code.
+        """
 
-        code = request.query_params.get('code')
         sku = request.query_params.get('sku')
+        code = request.query_params.get('code')
 
-        if not code:
-            return Response({'error': 'Code not provided.'})
+        if not request.user.id or not request.user.email:
+            raise PermissionDenied(detail="Could not detect user id or email.")
```

Comment:
Are both required, or just one?  Assuming both are required, consider tweaking the exception detail to be slightly clearer -- at least to me it reads like one is required but neither are present.

Reply:
Good point, will amend.

Reply:
Done: https://github.com/edx/commerce-coordinator/commit/76ba447a34ceedbf2c97d8523656f2ae8bb4f9a0